Os/test/ut/file/MyRules.cpp

```diff
@@ -0,0 +1,1150 @@
+        ASSERT_EQ(std::vector<U8>(read_data, read_data + size_read), state_data);
+        FwSignedSizeType position = -1;
+        FwSignedSizeType shadow_position = -1;
+        ASSERT_EQ(this->m_file.position(position), Os::File::Status::OP_OK);
+        ASSERT_EQ(this->m_shadow.position(shadow_position), Os::File::Status::OP_OK);
+        ASSERT_EQ(position, shadow_position);
+    }
+}
+
+void Os::Test::File::Tester::assert_file_write(const std::vector<U8>& write_data, FwSignedSizeType size_written) {
+    ASSERT_EQ(size_written, write_data.size());
+    FwSignedSizeType file_size = 0;
+    FwSignedSizeType shadow_size = 0;
+    ASSERT_EQ(this->m_file.size(file_size), Os::File::Status::OP_OK);
+    ASSERT_EQ(this->m_shadow.size(shadow_size), Os::File::Status::OP_OK);
+    ASSERT_EQ(file_size, shadow_size);
+    FwSignedSizeType file_position = -1;
+    FwSignedSizeType shadow_position = -1;
+    ASSERT_EQ(this->m_file.position(file_position), Os::File::Status::OP_OK);
+    ASSERT_EQ(this->m_shadow.position(shadow_position), Os::File::Status::OP_OK);
+    ASSERT_EQ(file_position, shadow_position);
+}
+
+void Os::Test::File::Tester::assert_file_seek(const FwSignedSizeType original_position, const FwSignedSizeType seek_desired, const bool absolute) {
+    FwSignedSizeType new_position = 0;
+    FwSignedSizeType shadow_position = 0;
+
+    ASSERT_EQ(this->m_file.position(new_position), Os::File::Status::OP_OK);
+    ASSERT_EQ(this->m_shadow.position(shadow_position), Os::File::Status::OP_OK);
+
+    const FwSignedSizeType expected_offset = (absolute) ? seek_desired : (original_position + seek_desired);
+    if (expected_offset > 0) {
+        ASSERT_EQ(new_position, expected_offset);
+    } else {
+        ASSERT_EQ(new_position, original_position);
+    }
+    ASSERT_EQ(new_position, shadow_position);
+}
+
+// ------------------------------------------------------------------------------------------------------
+//  OpenFile: base rule for all open rules
+//
+// ------------------------------------------------------------------------------------------------------
+Os::Test::File::Tester::OpenBaseRule::OpenBaseRule(const char *rule_name,
+                                                   Os::File::Mode mode,
+                                                   const bool overwrite,
+                                                   const bool randomize_filename)
+        : STest::Rule<Os::Test::File::Tester>(rule_name), m_mode(mode),
+          m_overwrite(overwrite ? Os::File::OverwriteType::OVERWRITE : Os::File::OverwriteType::NO_OVERWRITE),
+          m_random(randomize_filename) {}
+
+bool Os::Test::File::Tester::OpenBaseRule::precondition(const Os::Test::File::Tester &state  //!< The test state
+) {
+    return state.m_mode == Os::File::Mode::OPEN_NO_MODE;
+}
+
+void Os::Test::File::Tester::OpenBaseRule::action(Os::Test::File::Tester &state  //!< The test state
+) {
+    printf("--> Rule: %s \n", this->name);
```

Comment:
Why are we printing out rules manually instead of using STest to do this? Is the issue that we want the rule printed every time it is applied, instead of just in a random scenario? If so, should we update STest? The manual printing seems awkward.

Reply:
This pattern is from @gjwatney's rule generator.  It starts each rule with this.  With the new update to STest changing the name of rules to `.getName()` this breaks.  I will scrub them.

Would be nice to have for debug output from STest though.